gdal/ogr/ogrpolyhedralsurface.cpp

```diff
@@ -0,0 +1,857 @@
+
+        OGRSurface* poSurface;
+
+    /* -------------------------------------------------------------------- */
+    /*      Do the import.                                                  */
+    /* -------------------------------------------------------------------- */
+        if (EQUAL(szToken,"("))
+        {
+            OGRPolygon      *poPolygon = new OGRPolygon();
+            poSurface = poPolygon;
+            pszInput = pszInputBefore;
+            eErr = poPolygon->importFromWKTListOnly( (char**)&pszInput, bHasZ, bHasM,
+                                                     paoPoints, nMaxPoints, padfZ );
+        }
+        else if (EQUAL(szToken, "EMPTY") )
+        {
+            poSurface = new OGRPolygon();
+        }
+
+        /* We accept POLYGON() but this is an extension to the BNF, also */
+        /* accepted by PostGIS */
+        else if ((EQUAL(szToken,"POLYGON") ||
+                  EQUAL(szToken,"CURVEPOLYGON")))
+        {
+            OGRGeometry* poGeom = NULL;
+            pszInput = pszInputBefore;
+            eErr = OGRGeometryFactory::createFromWkt( (char **) &pszInput,
+                                                       NULL, &poGeom );
+            poSurface = (OGRSurface*) poGeom;
+        }
+        else
+        {
+            CPLError(CE_Failure, CPLE_AppDefined, "Unexpected token : %s", szToken);
+            eErr = OGRERR_CORRUPT_DATA;
+            break;
+        }
+
+        if( eErr == OGRERR_NONE )
+            eErr = oMP.addGeometryDirectly( poSurface );
+        if( eErr != OGRERR_NONE )
+        {
+            delete poSurface;
+            break;
+        }
+
+        // Read the delimiter following the surface.
+        pszInput = OGRWktReadToken( pszInput, szToken );
+
+    } while( szToken[0] == ',' && eErr == OGRERR_NONE );
+
+    CPLFree( paoPoints );
+    CPLFree( padfZ );
+
+    // Check for a closing bracket
+    if( eErr != OGRERR_NONE )
+        return eErr;
+
+    if( szToken[0] != ')' )
+        return OGRERR_CORRUPT_DATA;
+
+    *ppszInput = (char *) pszInput;
+    return OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                            exportToWkt()                             */
+/*      Translate this structure into it's well known text format       */
+/*      equivalent.                                                     */
+/************************************************************************/
+
+OGRErr OGRPolyhedralSurface::exportToWkt ( char ** ppszDstText,
+                                           OGRwkbVariant eWkbVariant ) const
+{
+    return exportToWktInternal(ppszDstText, eWkbVariant, "POLYGON");
+}
+
+OGRErr OGRPolyhedralSurface::exportToWktInternal ( char ** ppszDstText,
+                                                   OGRwkbVariant eWkbVariant,
+                                                   const char* pszSkipPrefix ) const
+{
+    char        **papszGeoms;
```

Comment:
Ignore the actual value of eWkbVariant in this method and consider it to be wkbVariantIso